lib/puppet/trusted_external.rb

```diff
@@ -3,11 +3,35 @@ module Puppet::TrustedExternal
   def retrieve(certname)
     command = Puppet[:trusted_external_command]
     return nil unless command
+    unless Puppet::FileSystem.directory?(command)
+      return fetch_data(command, certname)
```

Comment:
It could be neither a file nor directory. Suggest checking `if P::F.file?(command)` and calling `fetch_data` if so.

Reply:
Or you could rely on the setting's type, e.g. `Puppet.settings.setting[:trusted_external_command].type`, which calls https://github.com/puppetlabs/puppet/blob/master/lib/puppet/settings/file_or_directory_setting.rb#L7-L12

Reply:
Thanks, updated to use `settings.setting(...).type`